### DISCUSSION TITLE
Is there any Quick way, to test all endpoint

### DISCUSSION THREAD

**Discussion by Papazy** [fastapi/fastapi#11259] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Union

from fastapi import FastAPI, HTTPException, Depends, status
from fastapi.responses import RedirectResponse
from sqlalchemy.orm import Session
from .routers import problem_test_case_results, problem_test_cases
from .routers import users, problems, admins, languages, submissions, categories, contests, contest_participants, contest_problems, notification
from .database import engine, SessionLocal
from typing import Annotated
from pydantic import BaseModel

from .routers.local import local_test_cases, local_test_case_results , local_contests, local_contest_participants, local_contest_problems
from fastapi.middleware.cors import CORSMiddleware


app = FastAPI()

origins = ["*"]
app.add_middleware(
    CORSMiddleware,
    allow_origins=origins,
    allow_credentials=True,
    allow_methods=["POST", "GET", "PUT", "DELETE"],
    allow_headers=["*"],
)
@app.get('/', tags=["Landing Path"])
def redirect():
    return RedirectResponse('/docs')

# router 
app.include_router(users.router)
app.include_router(problems.router)
app.include_router(admins.router)
app.include_router(submissions.router)
app.include_router(languages.router)
app.include_router(categories.router)
app.include_router(contests.router)
app.include_router(contest_participants.router)
app.include_router(contest_problems.router)
app.include_router(problem_test_cases.router)
app.include_router(problem_test_case_results.router)
app.include_router(notification.router)

app.include_router(local_test_cases.router)
app.include_router(local_test_case_results.router)
app.include_router(local_contests.router)
app.include_router(local_contest_participants.router)
app.include_router(local_contest_problems.router)
```

### Description

Hello, i have so many endpoints to test. is there a quick way to test everything? for example, testing all endpoints result in 200. 

I just don't want to write all the tests one by one, because there are already so many endpoints

### Operating System

Windows

### Operating System Details

I'm using windows 11

### FastAPI Version

0.109.0

### Pydantic Version

2.5.3

### Python Version

3.11.6

**Comment by YraganTron** [comment]
Hi! You can find all files with routers and after you can get all endpoints. For example, you can do some function which get all endpoints and use pytest parametrize for test all your endpoints

**Comment by YuriiMotov** [comment, marked as answer]
I think you can use the AI assistant to create basic tests.
